src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const clone = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(clone, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete clone[key];
+        }
+        break;
+
+      case 'clear':
+        for (const del in clone) {
+          delete clone[del];
+        }
+        break;
+    }
+    result.push({ ...clone });
```

Comment:
Follow the checklist, please. Add a default case 
<img width="784" alt="image" src="https://user-images.githubusercontent.com/99403198/231425608-0a594de6-960c-4faf-97e6-9bf1027c00e9.png">